tests/test_errors.py

```diff
@@ -46,8 +46,19 @@ def test_no_tables_found():
     # TODO: use pytest.warns
     with warnings.catch_warnings():
         warnings.simplefilter('error')
-        try:
+        with pytest.raises(UserWarning) as e:
             tables = camelot.read_pdf(filename)
-        except Exception as e:
-            assert type(e).__name__ == 'UserWarning'
-            assert str(e) == 'No tables found on page-1'
+        assert str(e.value) == 'No tables found on page-1'
```

Comment:
@jonathanlloyd Should this be one indent level down?

Reply:
Its a bit weird but that's actually how it's documented: https://docs.pytest.org/en/latest/assert.html
It seems to work 🤷‍♂️

Reply:
Oh. I looked at the tests for requests and they seem to do it this way too. https://github.com/requests/requests/blob/master/tests/test_requests.py#L817 If you ever find why pytest implemented it this way, do comment here!

Reply:
Very odd!